### ISSUE TITLE
Data.Set.Internal error message says Map

### ISSUE THREAD

**Issue by erikd** [haskell/containers#902]
The file `containers/src/Data/Set/Internal.hs` in function `balanceR` has the following error:
```
       (_, _) -> error "Failure in Data.Map.balanceR"
```
where the "Map" part should be "Set".

I would have submitted a PR, but I got an odd git error when I tried to commit the fix.

**Comment by erikd** [comment]
```
diff --git a/containers/src/Data/Set/Internal.hs b/containers/src/Data/Set/Internal.hs
index 6e4d2d6c..84439afc 100644
--- a/containers/src/Data/Set/Internal.hs
+++ b/containers/src/Data/Set/Internal.hs
@@ -1764,7 +1764,7 @@ balanceR x l r = case l of
                    (Bin rls rlx rll rlr, Bin rrs _ _ _)
                      | rls < ratio*rrs -> Bin (1+ls+rs) rx (Bin (1+ls+rls) x l rl) rr
                      | otherwise -> Bin (1+ls+rs) rlx (Bin (1+ls+size rll) x l rll) (Bin (1+rrs+size rlr) rx rlr rr)
-                   (_, _) -> error "Failure in Data.Map.balanceR"
+                   (_, _) -> error "Failure in Data.Set.balanceR"
               | otherwise -> Bin (1+ls+rs) x l r
 {-# NOINLINE balanceR #-}
```

**Comment by erikd** [comment]
```
diff --git a/containers/src/Data/Set/Internal.hs b/containers/src/Data/Set/Internal.hs
index 6e4d2d6c..d5d7e29d 100644
--- a/containers/src/Data/Set/Internal.hs
+++ b/containers/src/Data/Set/Internal.hs
@@ -1739,7 +1739,7 @@ balanceL x l r = case r of
                    (Bin lls _ _ _, Bin lrs lrx lrl lrr)
                      | lrs < ratio*lls -> Bin (1+ls+rs) lx ll (Bin (1+rs+lrs) x lr r)
                      | otherwise -> Bin (1+ls+rs) lrx (Bin (1+lls+size lrl) lx ll lrl) (Bin (1+rs+size lrr) x lrr r)
-                   (_, _) -> error "Failure in Data.Map.balanceL"
+                   (_, _) -> error "Failure in Data.Set.balanceL"
               | otherwise -> Bin (1+ls+rs) x l r
 {-# NOINLINE balanceL #-}
 
@@ -1764,7 +1764,7 @@ balanceR x l r = case l of
                    (Bin rls rlx rll rlr, Bin rrs _ _ _)
                      | rls < ratio*rrs -> Bin (1+ls+rs) rx (Bin (1+ls+rls) x l rl) rr
                      | otherwise -> Bin (1+ls+rs) rlx (Bin (1+ls+size rll) x l rll) (Bin (1+rrs+size rlr) rx rlr rr)
-                   (_, _) -> error "Failure in Data.Map.balanceR"
+                   (_, _) -> error "Failure in Data.Set.balanceR"
               | otherwise -> Bin (1+ls+rs) x l r
 {-# NOINLINE balanceR #-}
```

**Comment by treeowl** [comment]
Thanks. Do you think you could open a PR?

**Comment by erikd** [comment]
I would love to open a PR, but I cannot even commit the fix:
```
> git-commit -m "Fix error message" -- .
error: 'containers/docs/_extensions/haddock-autolink' does not have a commit checked out
fatal: updating files failed
```

**Comment by treeowl** [comment]
Paging @m-renaud.

**Comment by treeowl** [comment]
This might be a git submodule thing...

**Comment by erikd** [comment]
Had to remember how to use submodules. PR coming.

**Comment by erikd** [comment]
https://github.com/haskell/containers/pull/903

**Comment by konsumlamm** [comment]
Fixed by #903, so this issue can be closed.
